buffer/src/main/java/io/netty5/buffer/adapt/AdaptivePoolingAllocator.java

```diff
@@ -518,17 +523,31 @@ public Buffer allocate(int size, int sizeBucket) {
                 if (curr.capacity() < RETIRE_CAPACITY) {
                     curr.close();
                     current = buffer;
-                } else if (!(boolean) NEXT_IN_LINE.compareAndSet(this, null, buffer)) {
-                    if (!parent.offerToQueue(buffer)) {
-                        // Next-in-line is occupied AND the central queue is full.
-                        // Rare that we should get here, but we'll only do one allocation out of this chunk, then.
-                        buffer.close();
-                    }
+                } else {
+                    transferChunk(buffer);
                 }
             }
             return result;
         }
 
+        private void transferChunk(Buffer current) {
+            if (NEXT_IN_LINE.compareAndSet(this, null, current)
+                    || parent.offerToQueue(current)) {
+                return;
+            }
+
+            Buffer nextChunk = NEXT_IN_LINE.get(this);
+            if (current.capacity() > nextChunk.capacity()) {
```

Comment:
I think we should add a null check
```suggestion
            if (nextChunk != null && current.capacity() > nextChunk.capacity()) {
```

Reply:
right, I've added it